server/src/main/java/org/opensearch/action/get/TransportGetAction.java

```diff
@@ -88,16 +92,32 @@ protected boolean resolveIndex(GetRequest request) {
         return true;
     }
 
+    static boolean isSegmentReplicationEnabled(ClusterState state, String indexName) {
+        return Optional.ofNullable(state.getMetadata().index(indexName))
+            .map(
+                indexMetadata -> ReplicationType.parseString(indexMetadata.getSettings().get(IndexMetadata.SETTING_REPLICATION_TYPE))
+                    .equals(ReplicationType.SEGMENT)
+            )
+            .orElse(false);
+    }
+
+    /**
+     * Returns true if GET request should be routed to primary shards, else false.
+     */
+    protected static boolean isPrimaryBasedRouting(ClusterState state, boolean realtime, String preference, String indexName) {
```

Comment:
Nitpick, `forcePrimaryBasedRouting` or `shouldForcePrimaryRouting` or similar is probably better.